source/common/quic/envoy_quic_client_stream.cc

```diff
@@ -93,6 +93,11 @@ Http::Status EnvoyQuicClientStream::encodeHeaders(const Http::RequestHeaderMap&
       (Http::HeaderUtility::isCapsuleProtocol(headers) ||
        Http::HeaderUtility::isConnectUdp(headers))) {
     useCapsuleProtocol();
+    if (Http::HeaderUtility::isConnectUdp(headers)) {
+      // HTTP/3 Datagrams sent over CONNECT-UDP are already congestion controlled, so makes it
```

Comment:
nits: s/makes/make.

Is it possible to have two requests with `Upgrade: Connect-UDP` header?

Reply:
Fixed the comments.

Are you asking if the CONNECT-UDP upgrade requests can be sent twice? It works same as the CONNECT, where after the first successful CONNECT-UDP request and response, the following messages must be all HTTP Datagrams (using the Capsule Protocol).

Reply:
Is there code to enforce this?

Reply:
Good point, currently it just creates a new HttpDatagramHandler as if a new CONNECT-UDP stream is established. I think I should reset the stream if the capsule protocol is already in use? Let me know if you have a different idea.

Can I send another PR to fix this issue separately and just merge this one first?

Reply:
SG